### PR TITLE
LPS-62533 DoS on ResourcePermission records using ppid

### PR DESCRIPTION
**Pull request by topolik**
Hi Brian,

https://issues.liferay.com/browse/LPS-62533

**Problem**: Chicken and egg :) 

To check permissions we need to add them. So we add them during the first time render, however we need them before the render. So we add them every time somebody requests them => changing ppid in URL will create any number of records in DB.

**Solution:** Create a "default" company-wide set of ResourcePermission records for a portlet.

When we place a portlet on a page we add a default set of ResourcePermission records. If we place a portlet 20 times on the same page, there will be 20 sets with the same records (only primKey differs).

So instead of creating the set for every portlet instance we use the default records.

When admin changes permissions of a portlet on page then a new set of permissions is created, as it was before.

**TL;DR**: Before - every portlet on a page had own set of ResourcePermission records. After this pull - portlet has own set of ResourcePermission records only when they are different from the default set.

/cc @JorgeFerrer

### CODE REVIEW

**Comment by brianchandotcom** [comment]
@topolik can you review my changes in https://github.com/topolik/liferay-portal/pull/275 and resend? Thx.

Also, what's a "root model resource". We have root portlet id, that makes sense, but the name root model confuses me. Can you rename that? Thx.

**Comment by brianchandotcom** [comment]
@topolik Jorge explained to me some more about the "rootModelResource". Can you see if we called it that anywhere in the code or in documentation in 6.2.x? If so, just keep the name. If not, see if you can come up with a better name. Thx @topolik

**Comment by JorgeFerrer** [comment]
Hey Brian,

After our conversation I've noticed that in the XML we do have the concept of "root" associated to a model resource. See for example the following snipper:
```
	<model-resource>
		<model-name>com.liferay.portlet.messageboards</model-name>
		<portlet-ref>
			<portlet-name>com_liferay_message_boards_web_portlet_MBAdminPortlet</portlet-name>
			<portlet-name>com_liferay_message_boards_web_portlet_MBPortlet</portlet-name>
		</portlet-ref>
		<root>true</root>
		<weight>1</weight>
```

**Comment by topolik** [comment]
Please see https://github.com/brianchandotcom/liferay-portal/pull/35428. Thanks.
